### PR TITLE
fix(interval, timeout): add cancel to $interval and $timeout service when DOM destroy

### LINKED ISSUE #46: $interval.cancel dose not working
$interval.cancel dose not working, so $interval service still alive when come back with router.

I think it is necessary that $interval.cancel when DOM destroy. And $timeout too.

### PR DESCRIPTION
**Pull request by koihatano**
Closes #46

### CODE REVIEW

File: angular-dfp.js [hunk 1]
```diff
@@ -429,10 +427,18 @@ angular.module('ngDfp', [])
...
               }
 
-              $timeout(function () {
+              timeoutPromise = $timeout(function () {
                 DoubleClick.refreshAds(id);
               }, scope.timeout);
             });
+
+            // Cancel $interval and $timeout service when DOM destroy
+            element.on('$destroy', function() {
```
Comment (ianmurrays) [inline angular-dfp.js]: I think this should be `scope.on`

Comment (koihatano) [inline angular-dfp.js]: Actually, scope.$on is working too.

So, According to reference, use element.on at directive, use $scope.$on at controller meybe.
https://docs.angularjs.org/api/ng/service/$interval

Comment (ianmurrays) [inline angular-dfp.js]: Yeah I feel like `element.on` is more of a propagation of the angular `$destroy` event on the DOM element itself, so `scope.$on` feels more correct.

**Comment by ianmurrays** [comment]
This is a nice catch, if you could change the line we were discussing (as long as it works), we can get this merged 😄

**Comment by koihatano** [comment]
I fixed it 😄

**Pull request merged**